sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -60,6 +111,46 @@ export class AssessmentController {
     return job.complete(outputVariables);
   }
 
+  /**
+   * Updates the assessment dynamic data if it was not updated already.
+   */
+  @ZeebeWorker("save-assessment-data", {
+    fetchVariable: [ASSESSMENT_ID, ASSESSMENT_DATA],
+  })
+  async saveAssessmentData(
+    job: Readonly<
+      ZeebeJob<SaveAssessmentDataJobInDTO, ICustomHeaders, IOutputVariables>
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    await this.studentAssessmentService.updateAssessmentData(
+      job.variables.assessmentId,
+      job.variables.assessmentData,
+    );
+    return job.complete();
+  }
+
+  /**
+   * Updates the NOA (Notice of Assessment) status if not updated yet.
+   */
+  @ZeebeWorker("update-noa-status", {
+    fetchVariable: [ASSESSMENT_ID],
+  })
+  async updateNOAStatus(
+    job: Readonly<
+      ZeebeJob<
+        UpdateNOAStatusJobInDTO,
+        UpdateNOAStatusHeaderDTO,
+        IOutputVariables
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    await this.studentAssessmentService.updateNOAApprovalStatus(
+      job.variables.assessmentId,
+      job.customHeaders.status,
```

Comment:
What is the need of custom header in this case?

Reply:
It tells the workers with status must be set. Since it is a static value it was passed as a header. The alternative would be to pass it as an input variable, which is also possible but I just thought that was not needed for this worker.
![image](https://user-images.githubusercontent.com/61259237/196822372-69f896e5-fcec-44da-89a3-f16fdcdb69a0.png)

Reply:
Yeah remember that from previous conversation. 👍